cirq-google/cirq_google/devices/serializable_device.py

```diff
@@ -108,6 +108,25 @@ def __init__(
         """
         self.qubits = qubits
         self.gate_definitions = gate_definitions
+        self._metadata = cirq.GridDeviceMetadata(
+            [
+                (pair[0], pair[1])
+                for gate_defs in gate_definitions.values()
+                for gate_def in gate_defs
+                if gate_def.number_of_qubits == 2
```

Comment:
Just wondering: this is just a defensive check right? It should never be false?

Reply:
Not sure, I just copied this from lower down in the module `qid_pairs` function.